src/App.jsx

```diff
@@ -47,17 +28,22 @@ function Itemslist(props) {
 Itemslist.propTypes = {
   className: PropTypes.string.isRequired,
   list: PropTypes.arrayOf(PropTypes.string).isRequired,
+  handelDelete: PropTypes.func.isRequired,
 };
 
 function App() {
-  const [itemsarray, setItemsarray] = useState([]);
+  const [itemsArray, setitemsArray] = useState([]);
   const [inputvalue, setInputvalue] = useState('');
 
+  const handelDelete = (id) => {
+    const newList = [...itemsArray].filter((item) => item.id !== id);
```

Comment:
This is not wrong, but `filter` function already copies the array and gives you a filtered copy, so you don't need to copy it twice.